lib/sisimai/bite/email/x3.rb

```diff
@@ -0,0 +1,141 @@
+module Sisimai::Bite::Email
+  module X3
+    # Sisimai::Bite::Email::X3 parses a bounce email which created by Unknown
+    # MTA #3. Methods in the module are called from only Sisimai::Message.
+    class << self
+      # Imported from p5-Sisimail/lib/Sisimai/Bite/Email/X3.pm
+      require 'sisimai/bite/email'
+
+      Re0 = {
+        :from    => %r/\AMail Delivery System/,
+        :subject => %r/\ADelivery status notification/,
+      }
+      Re1 = {
+        :begin  => %r/\A[ \t]+This is an automatically generated Delivery Status Notification/,
+        :rfc822 => %r|\AContent-Type: message/rfc822|,
+        :endof  => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'Unknown MTA #3'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return []; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from Unknown MTA #3
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
```

Comment:
__[rubocop]__
* Style/Semicolon: Do not use semicolons to terminate expressions. [:ref](https://github.com/bbatsov/ruby-style-guide#no-semicolon)
* Layout/TrailingWhitespace: Trailing whitespace detected. [:ref](https://github.com/bbatsov/ruby-style-guide#no-trailing-whitespace)

<!-- Comment by [SideCI](https://sideci.com) -->